### ISSUE TITLE
`styled` fn with Solid.js components

### ISSUE THREAD

**Issue by cschroeter** [chakra-ui/ark#2151]
### Description

This could be a Zag.js issue or the `styled` fn but using styled to wrap a `Select.Root` it return this error:

```test
TypeError: 'set' on proxy: trap returned truish for property 'items' which exists in the proxy target as a non-configurable and non-writable accessor property without a setter
```

```jsx
export const Demo = () => {
  const items = [
    { label: 'React', value: 'react' },
    { label: 'Solid', value: 'solid' },
    { label: 'Svelte', value: 'svelte', disabled: true },
    { label: 'Vue', value: 'vue' },
  ];
  return (
    <Select.Root items={items}>  {/* Using unstyled root works just fine */}
      {/* <Select.StyledRoot items={items}> This breaks with the error shown on top */}
      <Select.Label>Framework</Select.Label>
        <Select.Control>{/* ... */}</Select.Control>
        <Select.Positioner>{/* ... */}</Select.Positioner>
      {/* </Select.StyledRoot> */}
    </Select.Root>
  );
};
```

### Link to Reproduction

https://stackblitz.com/edit/github-rwgrxp?file=src%2Fapp.tsx,src%2Fcomponents%2Fui%2Fselect.tsx,src%2Fcomponents%2Fframework-select.tsx

**Comment by cschroeter** [comment]
@segunadebayo 

Can confirm that Storybook (Vite) works just fine.

The issue only occurs if the`styled` funcion from Panda is used. Maybe it is a combination of some factors. I'm not sure what I should do in Ark with this issue.

**Comment by asterikx** [comment]
After upgrading to `@pandacss/dev@0.30.2` I still get the same error. I'm not sure if the new release should fix this error or only https://github.com/chakra-ui/panda/issues/2156.

**Comment by asterikx** [comment]
I didn't see Solid's error overlay before (because of my custom error boundary) - it gives away a lot more: the seems to be related to the `toArray` function in Zag.

![localhost_3000_](https://github.com/chakra-ui/ark/assets/16797665/0345efc1-b12c-4e7f-9af1-9cd4635aa0e4)

**Comment by cschroeter** [comment]
The current `as` prop in Solid is not ideal. The previous example has multiple issues, mostly related to merging multiple types (ButtonProps + DialogTriggerProps + HTMLButtonElement), leading to a slow and, in some cases, non-functional current solution.

```jsx
// Before
<Button as={Dialog.Trigger} variant="solid" size="sm">
  Open Dialog
</Button>

// After
<Dialog.Trigger asChild>
  {(props) => (
    <Button
      {...props({ onClick: () => console.log('merge events') })}
      variant="solid"
      size="md"
    >
      Open Dialog
    </Button>
  )}
</Dialog.Trigger>
```

Even though the new proposal is a bit more verbose, it actually works and solves all the issues mentioned above. Please also bear in mind that you can still use this approach to avoid the `asChild` setup altogether:

```jsx
// Alternative, good for Tailwind, Bootstrap, DaisyUI, etc.
<Dialog.Trigger class="btn">Open Dialog</Dialog.Trigger>

// Alternative for Panda, Stylex
<Dialog.Trigger class={button({ size: '2xl', variant: '' })}>Open Dialog</Dialog.Trigger>
```


This change is breaking and will be available in v3 for Solid.js

**Comment by cschroeter** [comment]
Will be resolved with #2331
